### ISSUE TITLE
[tsql] Including both ON UPDATE CASCADE and ON DELETE CASCADE causes PRS ("Found unparsable section")

### ISSUE THREAD

**Issue by medianick** [sqlfluff/sqlfluff#6339]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following SQL script is triggering the PRS error:

```sql
ALTER TABLE [HangFire].[JobParameter] ADD CONSTRAINT [FK_HangFire_JobParameter_Job] FOREIGN KEY([JobId])
REFERENCES [HangFire].[Job] ([Id])
ON UPDATE CASCADE
ON DELETE CASCADE
```

As SQL Server/T-SQL supports including both ON UPDATE CASCADE and ON DELETE CASCADE (see [here](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-table-column-definition-transact-sql?view=sql-server-ver16#syntax)), this should be parsable and able to be linted.

### Expected Behaviour

I expect this to be parsable.

### Observed Behaviour

This raises a PRS ("Found unparsable section") linting error at the second "ON" statement.

### How to reproduce

See provided SQL

### Dialect

`tsql`

### Version

3.2.3

### Configuration

.sqlfluff:

```ini
[sqlfluff]
dialect = tsql
exclude_rules = CP01,CP02,LT01,LT02,LT05,LT12,RF06

[sqlfluff:rules:references.quoting]
prefer_quoted_identifiers = True

[sqlfluff:rules:references.special_chars]
allow_space_in_identifier = True
additional_allowed_characters = .
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
